lms/models/grouping.py

```diff
@@ -0,0 +1,71 @@
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+from lms.db import BASE
+from lms.models import TimestampedModelMixin
+from lms.models.application_settings import ApplicationSettings
+
+MAX_GROUP_NAME_LENGTH = 25
+
+
+class Grouping(TimestampedModelMixin, BASE):
+    __tablename__ = "grouping"
+    __mapper_args__ = {"polymorphic_identity": "grouping", "polymorphic_on": "type"}
+    __table_args__ = (
+        sa.UniqueConstraint("application_instance_id", "authority_provided_id"),
+    )
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    application_instance_id = sa.Column(
+        sa.Integer(),
+        sa.ForeignKey("application_instances.id", ondelete="cascade"),
+        nullable=False,
+    )
+    application_instance = sa.orm.relationship("ApplicationInstance")
+
+    authority_provided_id = sa.Column(sa.UnicodeText(), nullable=False)
+
+    parent_id = sa.Column(
```

Comment:
`null` for courses. Will be the corresponding course_grouping.id for sections and groups.